tests/asgi/tests.py

```diff
@@ -58,6 +59,13 @@ async def test_get_asgi_application(self):
         # Allow response.close() to finish.
         await communicator.wait()
 
+    # Python's file API is not async compatible. A third-party library such
+    # as https://github.com/Tinche/aiofiles allows passing the file to
+    # FileResponse as an async interator. With a sync iterator
+    # StreamingHTTPResponse triggers a warning when iterating the file.
+    # assertWarnsMessage is not async compatible, so ignore_warnings for the
+    # test.
+    @ignore_warnings(category=Warning)
```

Comment:
> I would wrap with assertWarnsMessage() to avoid leaking warnings to test logs.

@felixxm Just to repeat here what we chatted about, I had to use `@ignore_warnings` here for the whole test because `assertWarnsMessage` isn't async compatible. Happy to take another way if there is one.